### ISSUE TITLE
Cannot update Phel

### ISSUE THREAD

**Issue by bentxt** [phel-lang/web-skeleton#1]
Because of an error with the 'build' command I try to update 'web-skeleton' to Phels latest release. This doesnt work.

Here is the original ticket with the build error: https://github.com/phel-lang/phel-lang/issues/668

And here is the concerning part:

> Just a note: I try to update to phel  ^0.12 in  web-skeletons composer.json  but I get the following depency error, so I'm not sure how to proceed 

```
vim composer.json
[I] ben@moonraker ~/h/p/web-skeleton (main)> composer install
No composer.lock file present. Updating dependencies to latest instead of installing from lock file. See https://getcomposer.org/install for more information.
Loading composer repositories with package information
Updating dependencies
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Root composer.json requires phel-lang/router ^0.1 -> satisfiable by phel-lang/router[v0.1.0].
    - phel-lang/router v0.1.0 requires phel-lang/phel-lang ^0.10 -> found phel-lang/phel-lang[v0.10.0, v0.10.1] but it conflicts with your root composer.json require (^0.12).
  ```

**Comment by Chemaclass** [comment]
The error says: `No composer.lock file present.` Can you try using `composer update`, @bentxt?

Anyway, this is a good point. Probably having the `composer.lock` would simplify this DX 🤔

**Comment by Chemaclass** [comment]
I have upgraded web-skeleton to use the latest phel:0.12, and I am able to reproduce this bug https://github.com/phel-lang/phel-lang/issues/668 
Great finding! I will check this out
![Screenshot 2024-03-16 at 14 35 56](https://github.com/phel-lang/web-skeleton/assets/5256287/2eba8d9d-8cd7-44f7-9ad3-0f616d720c1d)

**Comment by Chemaclass** [comment]
This seems to be an issue on the `phel build` command, I think in the `ProjectCompiler` class. You can bypass this error if you use the argument `--no-cache`, but the actual error when building using the cache needs some investigation. I will create a follow up bug issue in the phel-lang repo.

TL;DR: using `phel build --no-cache` will recreate the out without issues.
